ssz_serialization/types.nim

```diff
@@ -424,10 +424,13 @@ func isFixedSize*(T0: type): bool {.compileTime.} =
   elif T is array|HashArray:
     return isFixedSize(ElemType(T))
   elif T is object|tuple:
-    enumAllSerializedFields(T):
-      when not isFixedSize(FieldType):
-        return false
-    return true
+    when T.isCaseObject():
```

Comment:
It's tricky to explain why, but due to some subtle Nim internal details, `T.isCaseObject()` is handled in a different way within the compiler that is more prone to bugs. This is the main reason, I've used the `isCaseObject(T)` syntax within the library.

Reply:
adjusted